From 0db546b1c10dfe863a375ce54191a580f3d88221 Mon Sep 17 00:00:00 2001
From: berbascum <berbascum@ticv.cat>
Date: Mon, 22 Jul 2024 00:55:33 +0200
Subject: [PATCH] (build) Fix: error section mismatch ksymtab_regulator_stub_init

WARNING: modpost: Found 1 section mismatch(es).

To see full details build your kernel with:
'make CONFIG_DEBUG_SECTION_MISMATCH=y'

WARNING: vmlinux.o(___ksymtab+regulator_stub_init+0x0): Section mismatch in reference from the variable __ksymtab_regulator_stub_init to the function .init.text:regulator_stub_init()
The symbol regulator_stub_init is exported and annotated __init 
Fix this by removing the __init annotation of regulator_stub_init or drop the export.

---
 drivers/regulator/stub-regulator.c       | 2 +-
 include/linux/regulator/stub-regulator.h | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/regulator/stub-regulator.c b/drivers/regulator/stub-regulator.c
index 5533349e6ebc..cdff338e10ed 100644
--- a/drivers/regulator/stub-regulator.c
+++ b/drivers/regulator/stub-regulator.c
@@ -282,7 +282,7 @@ static struct platform_driver regulator_stub_driver = {
 	},
 };
 
-int __init regulator_stub_init(void)
+int regulator_stub_init(void)
 {
 	static int registered;
 
diff --git a/include/linux/regulator/stub-regulator.h b/include/linux/regulator/stub-regulator.h
index 1d2fe4516af3..de51a9ef9324 100644
--- a/include/linux/regulator/stub-regulator.h
+++ b/include/linux/regulator/stub-regulator.h
@@ -40,7 +40,7 @@ struct stub_regulator_pdata {
  * registration ordering must be controlled precisely.
  */
 
-int __init regulator_stub_init(void);
+int regulator_stub_init(void);
 
 #else
 
-- 
2.39.2

